core/domain/exp_services.py

```diff
@@ -949,36 +949,46 @@ def update_exploration(
 
     exploration = apply_change_list(exploration_id, change_list)
     _save_exploration(committer_id, exploration, commit_message, change_list)
-
     # Update summary of changed exploration.
-    update_exploration_summary(exploration.id)
+    update_exploration_summary(exploration.id, committer_id=committer_id)
 
 
-def create_exploration_summary(exploration_id):
+def create_exploration_summary(exploration_id, committer_id=None):
     """Create summary of an exploration and store in datastore."""
     exploration = get_exploration_by_id(exploration_id)
-    exp_summary = compute_summary_of_exploration(exploration)
+    exp_summary = compute_summary_of_exploration(exploration, committer_id)
     save_exploration_summary(exp_summary)
 
 
-def update_exploration_summary(exploration_id):
+def update_exploration_summary(exploration_id, committer_id=None):
     """Update the summary of an exploration."""
     exploration = get_exploration_by_id(exploration_id)
-    exp_summary = compute_summary_of_exploration(exploration)
+    exp_summary = compute_summary_of_exploration(exploration, committer_id)
     save_exploration_summary(exp_summary)
 
 
-def compute_summary_of_exploration(exploration):
+def compute_summary_of_exploration(exploration, committer_id=None):
     """Create an ExplorationSummary domain object for a given Exploration
-    domain object and return it.
+    domain object and return it. Committer_id will be added to the list
+    of contributors for the exploration if the id is not a system id.
     """
     exp_rights = exp_models.ExplorationRightsModel.get_by_id(exploration.id)
     exp_summary_model = exp_models.ExpSummaryModel.get_by_id(exploration.id)
     if exp_summary_model:
         old_exp_summary = get_exploration_summary_from_model(exp_summary_model)
         ratings = old_exp_summary.ratings or feconf.get_empty_ratings()
+        contributor_ids = old_exp_summary.contributor_ids or []
     else:
         ratings = feconf.get_empty_ratings()
+        contributor_ids = []
+
+    # update the contributor id list if necessary (contributors
+    # defined as humans who have made a positive (i.e. not just
+    # a revert) change to an exploration's content)
+    if (committer_id is not None and
```

Comment:
Makes sense re:boolean non-optional arg-- agreed that this would be clearer.

Where should I put that test? In exp_services_test.ExplorationGetTests?